pkg/summary/summary.go

```diff
@@ -109,6 +117,12 @@ func Summarize(runtimeObj runtime.Object) Summary {
 	}
 
 	conditions := getConditions(obj)
+	if opts != nil {
+		if summary.Attributes == nil {
+			summary.Attributes = make(map[string]interface{})
```

Comment:
nit: Not a fan of using summary.Attributes for this instead of a proper typed field.. Have you considered adding a new field (unexported perhaps) to summary?